File: js/solution_generator.js
// solution generator

function SolutionGenerator() {
  this.generate = function() {
    var colors = ["yellow", "blue", "green", "red", "purple", "orange"];
    var solution = [];

    while (solution.length < 4) {
      var pick = Math.floor(Math.random() * colors.length);
      solution.push(colors.splice(pick, 1)[0]);
    }

    return solution;
  };
};

File: ruby/solution_generator.rb
module SolutionGenerator
	
  def self.generate_solution
    colors = [:yellow, :blue, :green, :red, :cyan, :light_magenta]
    solution = []

    until solution.length == 4
      pick = colors.sample
      solution << colors.delete(pick)
    end

    solution
  end
end

# initiate different levels of difficulty
